src/currencymanager.cpp

```diff
@@ -0,0 +1,282 @@
+}
+
+int CurrencyManager::TotalWisdomValue()
+{
+    int out = 0;
+    for (unsigned int i=0; i < wisdoms_.size(); i++) {
+        out += wisdoms_[i] * CurrencyWisdomValue[i];
+    }
+    return out;
+}
+
+void CurrencyManager::ClearCurrency()
+{
+    for(auto &currency : currencys_) {
+        currency.count = 0;
+    }
+    for( auto &wisdom : wisdoms_) {
+        wisdom = 0;
+    }
+}
+
+void CurrencyManager::InitCurrency()
+{
+    std::string value = "";
+    for (auto curr : CurrencyAsString) {
+        value += "0;";
+    }
+    value.pop_back();//Remove the last ";"
+    data_.Set("currency_base", value, "data");
+    data_.Set("currency_last_value",value, "data");
+}
+
+void CurrencyManager::LoadCurrency()
+{
+    //Way easier to use QT function instead of re-implementing a split() in C++
+    QStringList list = QString(data_.Get("currency_base", "", "data").c_str()).split(";");
+    for (int i=0; i < list.size(); i++) {
+        CurrencyItem curr;
+        curr.count = 0;
+        curr.name = CurrencyAsString[i];
+        //We can't use the toDouble function from QT, because it encodes a double with a ","
+        //Might be related to localisation issue, but anyway it's safer this way
+        curr.base = std::stod(list[i].toStdString());
+        curr.exalt = 0;
+        currencys_.push_back(curr);
+    }
+    for (unsigned int i=0; i < CurrencyWisdomValue.size(); i++)
+    {
+        wisdoms_.push_back(0);
+    }
+
+}
+
+void CurrencyManager::SaveCurrencyBase()
+{
+    std::string value = "";
+    for (auto currency : currencys_) {
+        value += std::to_string(currency.base) + ";";
+    }
+    value.pop_back();//Remove the last ";"
+    data_.Set("currency_base", value, "data");
+}
+void CurrencyManager::SaveCurrencyValue()
+{
+    std::string value = "";
+    //Useless to save if every count is 0.
+    bool empty = true;
+    for (auto currency : currencys_) {
+        if (currency.name !="")
+            value += std::to_string(currency.count) + ";";
+        if (currency.count !=0)
+            empty = false;
+    }
+    value += std::to_string(TotalExaltedValue());
+    std::string old_value = data_.Get("currency_last_value", "", "data");
+    if (value != old_value && !empty){
+        std::string timestamp = std::to_string(std::time(nullptr));
+        data_.Set(timestamp, value, "currency");
+        data_.Set("currency_last_value", value, "data");
+    }
+}
+
+void CurrencyManager::ExportCurrency()
+{
+    std::string header_csv = "";
+    for (auto &name : CurrencyAsString) {
+        if (name != "")
+            header_csv += name + ";";
+    }
+    header_csv += "Total value;Timestamp";
+    std::vector<std::string> result = data_.GetAllCurrency();
+
+    QString dir = QFileDialog::getExistingDirectory(this, tr("Select Directory"), QDir::homePath(),
+                                                    QFileDialog::ShowDirsOnly);
+    QString location = QDir::toNativeSeparators(dir + "/acquisition_export_currency.csv");
```

Comment:
why don't you use a "save file" dialog instead? so the user can specify file name as well as the directory